src/sparseml/core/recipe/container.py

```diff
@@ -90,33 +93,43 @@ def update(
         :param kwargs: additional kwargs to return
         :return: the passed in kwargs
         """
-        if recipe is not None:
-            self.compiled_recipe = None
-
-            if not isinstance(recipe, list):
-                recipe = [recipe]
-            if recipe_stage is None:
-                recipe_stage = [None] * len(recipe)
-            else:
-                if not isinstance(recipe_stage, list):
-                    recipe_stage = [[recipe_stage]] * len(recipe)
-                if not isinstance(recipe_stage[0], list):
-                    recipe_stage = [recipe_stage] * len(recipe)
-
-            if recipe_args is None:
-                recipe_args = [{}] * len(recipe)
-            elif not isinstance(recipe_args, list):
-                recipe_args = [recipe_args] * len(recipe)
-
-            if len(recipe) != len(recipe_stage) or len(recipe) != len(recipe_args):
-                raise ValueError(
-                    "recipe, recipe_stage, and recipe_args must be the same length"
-                )
-
-            for rec, stage, args in zip(recipe, recipe_stage, recipe_args):
-                if isinstance(rec, str):
-                    rec = Recipe.create_instance(rec)
-                self.recipes.append(RecipeTuple(rec, stage, args))
+        if recipe is None or isinstance(recipe, list) and len(recipe) == 0:
+            return kwargs
+
+        # Recipe is Truthy
```

Comment:
???? This comment is a mystery

Reply:
I just meant the recipy is non none and not an empty list at this point, but maybe it makes sense to remove it

Reply:
Removed!